### DISCUSSION TITLE
[2023-03-13] Component Browser Team Update

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#5907] [Weekly Demo Day]
* The improvement of Component Browser's Language Understanding is almost complete. The suggestions are displayed for the currently edited name, and literals are handled properly. The only thing remaining is incorporating qualified names already written by the user.
* The resorting groups when filtering was merged. Now, the best matches should be handy, and the best match is always selected by default.

Fixes:
* Fixed visualization attaching right after opening project (instead of being attached after showing it).
* Fixed Duplicated entries being shown again
* Merged initialization retries. IDE should be more stable on cloud.
* An issue where the visualization scroll bar is displayed over visualization list is fixed.
* Fix for wrong node position after dropping a file is ready.

**Comment by farmaazon** [comment]
A screenshot of reordered groups after typing "trim":

![Screenshot from 2023-03-10 14-11-02](https://user-images.githubusercontent.com/3919101/224746821-197c2c68-e704-4b0a-8c06-360f83f8c98c.png)

**Comment by vitvakatu** [comment]
### Finishing issue #5220:

1. Refactored the code for recognizing text literals in the CB input. Now it supports all types of string delimiters. Also the documentation for `Text` and `Number` types is shown when the literal is recognized.
2. Finally fixed the annoying issue with list filtering when opening the CB.

### Developing issue #5356

Component Browser recognizes qualified names in the input field and produces correct code on accepting suggestions.

https://user-images.githubusercontent.com/6566674/224742043-734c9721-b1c6-4b28-a720-0801e23ef6ab.mp4

https://user-images.githubusercontent.com/6566674/224742056-d4a5eb14-f4ae-4fe3-858b-15501067a027.mp4

**Comment by MichaelMauderer** [comment]
### Nodes being created in the right place when dropping files #5237 

https://user-images.githubusercontent.com/1428930/225037444-6216cbdd-5604-42d6-8748-75c3b0b81df7.mp4

### Scrollbars no longer overlapping visualization chooser #5089 

https://user-images.githubusercontent.com/1428930/225037585-308c4c27-ade1-4754-aa75-077eef5b4281.mp4
